projects/components/src/viewer/code-viewer/code-viewer.component.ts

```diff
@@ -204,6 +205,31 @@ export class CodeViewerComponent implements AfterViewInit, OnChanges, OnDestroy
     });
   }
 
+  private isLineHighlighted(lineNum: number): boolean {
+    /**
+     * `isEmpty(new RexExp('anything'))` returns true, so here we need to make another check for RegExp
+     */
+    const emptyValue =
+      this.highlightText instanceof RegExp ? isEmpty(this.highlightText.source) : isEmpty(this.highlightText);
+    if (emptyValue) {
+      return false;
+    }
+
+    if (typeof this.highlightText === 'string') {
+      return this.codeTexts[lineNum].toLowerCase().includes(this.highlightText.toLowerCase());
+    }
+
+    if (Array.isArray(this.highlightText)) {
+      return this.highlightText.some(text => this.codeTexts[lineNum].toLowerCase().includes(text.toLowerCase()));
+    }
+
+    if (this.highlightText instanceof RegExp) {
+      return (this.codeTexts[lineNum].match(this.highlightText) ?? []).length > 0;
+    }
+
+    throw new Error('Invalid type for highlight text, only string, string[] and RegExp is allowed');
```

Comment:
Would it not break the ui? we should rather fail it gracefully

Reply:
I don't think so, and it there for the dev people, as they can see the error if they pass wrong values in here

Reply:
This is dependent on them doing the manual testing and verifying it. This could be easily miss and cause a regression imo. The problem is this would likely break the entire page/angular if this happens.  I would suggest that you remove the throw Error.